src/splice.ts

```diff
@@ -1,12 +1,104 @@
+import type {
+  ArraySplice,
+  GreaterThan,
+  IsNegative,
+  Subtract,
+  Sum,
+} from "type-fest";
+import type {
+  CoercedArray,
+  IterableContainer,
+  TupleParts,
+} from "./internal/types";
 import { purry } from "./purry";
 
+type FixedLengthSplice<
+  T extends IterableContainer,
+  Start extends number,
+  DeleteCount extends number,
+  Replacement extends IterableContainer,
+  PositiveStart extends number = IsNegative<Start> extends false
+    ? Start
+    : number extends T["length"]
+      ? // Can't compute negative start index.
+        number
+      : Sum<T["length"], Start> extends infer SumResult extends number
+        ? number extends SumResult
+          ? // type-fest Sum returns number when the sum is negative.
+            0
+          : SumResult
+        : never,
+  ClampedDeleteCount extends number = IsNegative<DeleteCount> extends true
+    ? 0
+    : number extends T["length"]
+      ? // Don't bother clamping in this case.
+        number
+      : GreaterThan<Sum<PositiveStart, DeleteCount>, T["length"]> extends true
+        ? Subtract<
+            T["length"],
+            PositiveStart
+          > extends infer SubtractResult extends number
+          ? number extends SubtractResult
+            ? // type-fest Subtract returns number when the difference is negative.
+              0
+            : SubtractResult
+          : never
+        : DeleteCount,
+> = ArraySplice<T, PositiveStart, ClampedDeleteCount, Replacement>;
+
+type Splice<
+  T extends IterableContainer,
+  Start extends number,
+  DeleteCount extends number,
+  Replacement extends IterableContainer,
+> = TupleParts<T>["item"] extends never
+  ? FixedLengthSplice<T, Start, DeleteCount, Replacement>
+  : IsNegative<Start> extends true
+    ? // type-fest Sum returns number when the sum is negative.
+      number extends Sum<TupleParts<T>["suffix"]["length"], Start>
+      ? // Splice cuts into the variable-length part.
+        T
```

Comment:
Do you think there should be a TODO here? This is why we have a known issue in the tests, right?

Reply:
i was thinking about this... i think we could just mark the suffix optional and it'd work? let me try